### PR TITLE
wallet: ensure sat/vB feerates are in range (mantissa of 3)

### LINKED ISSUE #20534: sat/b values aren't validated to be in-range
values smaller than 0.001 sat/B can't be represented by CFeeRate, but they are not rejected.

```
./src/bitcoin-cli -named sendtoaddress address=bcrt1qwkmhqum095zau5rf2velq55cq938vrd7uw8m4r amount=1.23 fee_rate=0.0009

error code: -6
error message:
Fee rate (0.000 sat/vB) is lower than the minimum fee rate setting (1.000 sat/vB)
```

```
./src/bitcoin-cli -named sendtoaddress address=bcrt1qwkmhqum095zau5rf2velq55cq938vrd7uw8m4r amount=1.23 fee_rate=0.000000000009

error code: -3
error message:
Invalid amount

### PR DESCRIPTION
**Pull request by jonatack**
- Improve/close gaps in existing test coverage before making the change
- Enable passing `decimals` to `ParseFixedPoint()` when calling `AmountFromValue()`
- Limit explicit fee rates in sat/vB passed in by users to 3 decimals, and raise otherwise
- Add regression test coverage

Closes #20534.

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -216,7 +216,8 @@ static void SetFeeEstimateMode(const CWallet& wallet, CCoinControl& cc, const Un
         if (!estimate_mode.isNull() && estimate_mode.get_str() != "unset") {
             throw JSONRPCError(RPC_INVALID_PARAMETER, "Cannot specify both estimate_mode and fee_rate");
         }
-        cc.m_feerate = CFeeRate(AmountFromValue(fee_rate), COIN);
+        // Pass a mantissa of 3 as sat/vB values < 0.001 cannot be represented.
```
Comment (jonatack) [inline src/wallet/rpcwallet.cpp:219]:
```suggestion
        // Pass a mantissa of 3 as sat/vB values cannot represent values having more than 3 significant digits.
```

**Comment by MarcoFalke** [comment]
Concept ACK

**Comment by jonatack** [comment]
rebased

**Comment by jonatack** [comment]
CI failure is unrelated (https://github.com/bitcoin/bitcoin/issues/21632).

**Comment by jonatack** [comment]
Rebased for CI fix on master.

**Comment by meshcollider** [comment]
Concept ACK

**Review by fjahr — COMMENTED** [review]
Code review ACK 9c7b474cfd71cc5311db4e98c0efd1736a1096fb

modulo I think those two tests mentioned below can be removed now.

File: test/functional/rpc_fundrawtransaction.py [hunk 2]
```diff
@@ -770,6 +770,9 @@ def test_option_feerate(self):
                 node.fundrawtransaction, rawtx, {param: {"foo": "bar"}, "add_inputs": True})
             assert_raises_rpc_error(-3, "Invalid amount",
```
[the remark this answers is not in the document]
Comment (jonatack) [reply test/functional/rpc_fundrawtransaction.py:771]: Thanks! Done.

File: test/functional/rpc_psbt.py [hunk 3]
```diff
@@ -214,6 +214,10 @@ def run_test(self):
                 self.nodes[1].walletcreatefundedpsbt, inputs, outputs, 0, {param: {"foo": "bar"}, "add_inputs": True})
             assert_raises_rpc_error(-3, "Invalid amount",
```
[the remark this answers is not in the document]
Comment (jonatack) [reply test/functional/rpc_psbt.py]: Indeed, good eye. Done.

**Comment by jonatack** [comment]
Thank you @fjahr, updated with your feedback: `git diff 9c7b474 847288d`

**Comment by fjahr** [comment]
Code review ACK 9c7b474cfd71cc5311db4e98c0efd1736a1096fb

The only change was the removal of two now redundant/duplicate tests.

**Review by maflcko — APPROVED** [review]
review ACK 847288df07b45ca535c849e518b22818ab492896 🔷

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

review ACK 847288df07b45ca535c849e518b22818ab492896 🔷
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUjgOAv/U64jIK5O7lauqHgzSxVR8ARVhocyv95mL70s0DRDSUXyWeNjsQAbqq4d
d/Frd1URcN+ui/wiGAvzW7JjgPxdFBfUpcUZCh6uVwzMYIi5JfzDpVVC1Fa/S9PG
Wcvi14s/N7wWMEYRE85WMncceEyCAb5NP6/Pnr8zY3/jdMT+NGZv+OvIkNgpxpFt
CP9iMkX5GFuHOJnFc53PNAvvuGnoDE/jGWDk7X48Q8EScNgwKnk16K7vNDbarKb2
szmXO2H+ql7moWoTBK0xDr/X2GEdZw+49nr1aP11pxBAIbgk0T/PweW6HQlnKaQ4
Ys/mVrSS18laMZkK2LvOgHQynSHmwwUyekcCzaxNWkRP2Si17bIwWlCCswPL4L1Z
qgWDQomJ2eJoYUBJMFwxTe0ZqVlRNEQA3ah5dv9qmImo9Qezf28E5fHeZ/OmHiEo
dBECvDU5QZlwFsUZ94kRY7t0/8E+5yGDESsmYV3120BkOYLcGx+OARlT0YZcyRaz
7SjbaQL7
=gqY6
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `8b19ebaa4f993d9f7f5a3cd958924c1f48d2c00e94f5d00a4b4b7012058dc816  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401088b19ebaa4f993d9f7f5a3cd958924c1f48d2c00e94f5d00a4b4b7012058dc816f010a54c1cd9c3006846d61197ed3928895608fff010366bac85d8a478f2f9c64e2c0759609008f1046098da7bf00865feb4089d7265b30083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010373f8c5d163b8107d213eca336436d3508f1046098da7af008e015dfc23cd080c60083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6df010611edf9a985de3c5ba1495c7cdf421d008f1046098da7af008a7d5092d6272ba070083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

**Pull request merged**